lightning/src/onion_message/messenger.rs

```diff
@@ -175,6 +175,8 @@ where
 	message_router: MR,
 	offers_handler: OMH,
 	custom_handler: CMH,
+	intercept_oms_for_offline_peers: bool,
+	pending_events: Mutex<Vec<Event>>,
```

Comment:
Why use a separate queue if these aren't persisted? If we instead use `message_recipients` for buffering -- with a new variant -- then we can re-use `outbound_buffer_full` and generate events in the same manner as `ConnectionNeeded`. Otherwise, we are left with two different systems for buffering messages and generating events.

Reply:
Hmm, I believe with this approach an attacker may be able to fragment our heap by continually sending bogus forwards to a new fake peer each time (where each fake forward results in a `VecDeque` allocation). I think my other reasoning is that because the events generated may all be bogus, they shouldn't be allowed to count towards `outbound_buffer_full`, since currently all of that traffic is known to be more legitimate (either an outbound OM initiated by the LDK user or a forward to a peer that is connected).

Reply:
Discussed offline. Yeah, I suppose specifying which peers support forwarding adds a bit of user to the burden. They are already doing this when processing events, so no sense forcing them to duplicate it here an risk inconsistencies.